### PR TITLE
Toyota Pedal Support

### PR DESCRIPTION
**Pull request by wocsor**
Tested on 2018 Corolla LE and confirmed to work without Pedal on Prius Prime.
Scaling may need to be tuned for Pedal CAN input

### CODE REVIEW

**Comment by rbiasini** [comment]
@wocsor , can you make a pull request for opendbc with dbc file changes first? I can merge that right away.

**Comment by rbiasini** [comment]
@wocsor , can you make a separate Toyota Highlander pull request? I can merge that easily and would like to isolate pedal changes as much as possible.

**Comment by wocsor** [comment]
sorry, the highlander was just a test, it doesn't work. i can make a seperate PR for the DBC changes

**Comment by priusrav4** [comment]
can conform it works on RAV4 2016 SE!

**Comment by icmma** [comment]
Bro @rbiasini  can you please upstream pedal support confirmed working on almost all Toyota's.

**Comment by rbiasini** [comment]
allright, declared goal is to merge it by end of next week. See comments as I go through it.

File: panda/board/pedal.toyota/Makefile [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# :set noet
```
Comment (rbiasini) [inline panda/board/pedal.toyota/Makefile]: can the folder name be `pedal_toyota`?
I'll change `pedal` in `pedal_honda`.

**Comment by rbiasini** [comment]
I need to merge a PR in panda repo first. I'll start there if you can make one.

**Comment by wocsor** [comment]
I'll make the PR to Panda now

**Comment by rbiasini** [comment]
Panda done. opendbc next?

File: selfdrive/car/toyota/interface.py [hunk 2]
```diff
@@ -147,14 +154,13 @@ def get_params(candidate, fingerprint):
     ret.longPidDeadzoneBP = [0., 9.]
     ret.longPidDeadzoneV = [0., .15]
 
+    #detect the Pedal address
+    ret.enableGasInterceptor = 0x201 in fingerprint
+
+
```
Comment (rbiasini) [inline selfdrive/car/toyota/interface.py]: extra line

File: selfdrive/car/toyota/values.py [hunk 3]
```diff
@@ -79,7 +79,7 @@ def check_ecu_msgs(fingerprint, ecu):
...
+    36: 8, 37: 8, 170: 8, 180: 8, 186: 4, 426: 6, 452: 8, 464: 8, 466: 8, 467: 8, 512: 6, 513: 6, 547: 8, 548: 8, 552: 4, 562: 4, 608: 8, 610: 5, 643: 7, 705: 8, 725: 2, 740: 5, 800: 8, 835: 8, 836: 8, 849: 4, 869: 7, 870: 7, 871: 2, 896: 8, 897: 8, 900: 6, 902: 6, 905: 8, 911: 8, 916: 3, 918: 7, 921: 8, 933: 8, 944: 8, 945: 8, 951: 8, 955: 4, 956: 8, 979: 2, 998: 5, 999: 7, 1000: 8, 1001: 8, 1005: 2, 1008: 2, 1014: 8, 1017: 8, 1041: 8, 1042: 8, 1043: 8, 1044: 8, 1056: 8, 1059: 1, 1114: 8, 1161: 8, 1162: 8, 1163: 8, 1176: 8, 1177: 8, 1178: 8, 1179: 8, 1180: 8, 1181: 8, 1190: 8, 1191: 8, 1192: 8, 1196: 8, 1227: 8, 1228: 8, 1235: 8, 1237: 8, 1263: 8, 1264: 8, 1279: 8, 1408: 8, 1409: 8, 1410: 8, 1552: 8, 1553: 8, 1554: 8, 1555: 8, 1556: 8, 1557: 8, 1561: 8, 1562: 8, 1568: 8, 1569: 8, 1570: 8, 1571: 8, 1572: 8, 1584: 8, 1589: 8, 1592: 8, 1593: 8, 1595: 8, 1596: 8, 1597: 8, 1600: 8, 1656: 8, 1664: 8, 1728: 8, 1745: 8, 1779: 8, 1904: 8, 1912: 8, 1990: 8, 1998: 8
```
Comment (rbiasini) [inline selfdrive/car/toyota/values.py]: I assume those are all the cars where pedal has been verified?

File: selfdrive/car/toyota/interface.py [hunk 4]
```diff
@@ -234,7 +246,11 @@ def update(self, c):
 
     # gas pedal
     ret.gas = self.CS.car_gas
-    ret.gasPressed = self.CS.pedal_gas > 0
+    if self.CP.enableGasInterceptor:
+    # use interceptor values to disengage on pedal press
+      ret.gasPressed = self.CS.pedal_gas > 15
```
Comment (rbiasini) [inline selfdrive/car/toyota/interface.py]: what was 15 based on? Noise tolerance?

File: selfdrive/car/toyota/values.py (hunk 3 above)
Comment (wocsor) [reply selfdrive/car/toyota/values.py]: That's correct, it should work on all Toyotas that have a DSU to unplug

File: selfdrive/car/toyota/interface.py (hunk 4 above)
Comment (wocsor) [reply selfdrive/car/toyota/interface.py]: Yes. But the leeway is small enough that even a tap on the gas should still disengage

File: selfdrive/car/toyota/interface.py (hunk 2 above)
Comment (wocsor) [reply selfdrive/car/toyota/interface.py]: removed

**Comment by wocsor** [comment]
latest panda and opendbc added!

**Comment by rbiasini** [comment]
Merged internally. Will be on next release. My comments after driving on it for a few commutes (on RAV4):
- accel from stop is pretty good
- Needs tuning: starting at zero speed and enabling with no car in front creates an oscillatory acceleration behavior.
- would be nice to change the lookups so that > 19mph the controls smoothly switches back to the standard one and pedal isn't controlled anymore.

Overall, good job!

**Comment by arne182** [comment]
Tuning factors can be found on my fork, but I suggest splitting the RAV4 and RAV4H because they need to have different factors.

**Pull request closed without merging**